src/main/java/com/iexec/core/replicate/ReplicatesService.java

```diff
@@ -404,26 +483,25 @@ private boolean verifyStatus(String chainTaskId, String walletAddress, Replicate
         }
     }
 
-    private boolean updateReplicateWeight(String chainTaskId, Replicate replicate) {
+    private boolean canUpdateReplicateWeight(String chainTaskId,
```

Comment:
This method looks a little bit empty, cant we merge it in the calling method?

Reply:
I tend to prefer small methods as their name and Javadoc give precious information without having to read ~10 lines of code. Do you think it could be an issue?

Reply:
The method name doesn't express entirely what the method does (eg something like: areResutHashAndWorkerWeightPresent or more generic)

Reply:
If we go for a so-detailed method name, I think it would be better to separate both checks in new methods - `isResultHashPresent` and `isWorkerWeightPresent` - as they don't have anything in common but their presence for the replicate weight to be updated.
Would that suits you?

Reply:
Yes, I like that too.
Can you confirm we need to store the whole chainContribution in updateArgs? (required for resultHash and <?>)

Reply:
We could technically store only the `resultHash`. However, the current check is on the whole `chainContribution` object:
```
        if (chainContribution == null) {
            log.error("Failed to get chain contribution [chainTaskId:{}, workerWallet:{}]",
                    chainTaskId, walletAddress);
            return false;
        }
```
We could probably call our new check methods - `isWorkerWeightPresent` and `isChainContributionPresent` - in `computeUpdateReplicateStatusArgs` and throw exceptions in case of bad values but I don't see many Exceptions in our current base code. Do you consider it to be a code smell?
If we don't use exceptions there, I don't think we could tweak the behavior of the code to store only `resultHash` instead of the whole `chainContribution` object.